abe/fame.go

```diff
@@ -349,9 +384,25 @@ func (a *FAME) Decrypt(cipher *FAMECipher, key *FAMEAttribKeys, pk *FAMEPubKey)
 		ctPairing := bn256.Pair(ctProd[j], key.K0[j])
 		keyPairing := bn256.Pair(keyProd[j], cipher.Ct0[j])
 		keyPairing.Neg(keyPairing)
-		msgInGt.Add(msgInGt, ctPairing)
-		msgInGt.Add(msgInGt, keyPairing)
+		keyGt.Add(keyGt, ctPairing)
+		keyGt.Add(keyGt, keyPairing)
 	}
 
-	return bn256.MapGTToString(msgInGt), nil
+	keyCBC := sha256.Sum256([]byte(keyGt.String()))
+
+	c, err := aes.NewCipher(keyCBC[:])
+	if err != nil {
+		return "", err
+	}
+	iv := make([]byte, c.BlockSize())
+
+	msgPad := make([]byte, len(cipher.SymEnc))
+	decrypter := cbc.NewCBCDecrypter(c, iv)
+	decrypter.CryptBlocks(msgPad, cipher.SymEnc)
```

Comment:
Decrypter or decryptor?

Reply:
Actually  [wikipedia](https://en.wiktionary.org/wiki/decrypter#English) reports both versions, I used decrypter because go implementation of CBC also calls it so (see `NewCBCDecrypter`).

Reply:
Interesting. I was just attempting to be consistent across schemes (I think we usually use decryptor), but it's not really an issue.